native/tests/src/test_network.cpp

```diff
@@ -0,0 +1,78 @@
+#include <gtest/gtest.h>
+#include <mixpanel/mixpanel.hpp>
+#include <mixpanel/detail/worker.hpp>
+#include "../../source/dependencies/nano/include/nanowww/nanowww.h"
+#include "test_config.hpp"
+
+//
+// Note: We've placed these tests in to the mixpanel namespace so we can access
+//       private members and functions as a friend class. This makes use of
+//       the FRIEND_TEST macro in worker.hpp
+//
+namespace mixpanel
+{
+    namespace detail
+    {
+        // Ensure "Retry-After" HTTP header is respected
+        TEST(MixpanelNetwork, RetryAfter)
+        {
+            auto mp = new mixpanel::Mixpanel(mp_token);
+            auto worker = new mixpanel::detail::Worker(mp);
```

Comment:
Remove the `new` in both cases, or use `std::make_unique`, or something; anyway, please don't leak these allocated objects.